tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/chunk/ChunkReader.java

```diff
@@ -55,7 +55,7 @@
 
   protected Filter filter;
 
-  private List<IPageReader> pageReaderList = new LinkedList<>();
+  private List<IPageReader> pageReaderList = new ArrayList<>();
```

Comment:
In chunkReader Line 118, `return pageReaderList.remove(0).getAllSatisfiedPageData();`

Reply:
We think the following part actually performs random access upon the list object frequently in a loop. It would be quite expensive. Compare to the remove(0) you mentioned, I still think it would be better to change it to ArrayList. Remove(0) would invoke the copy of a successive memory, which can be conducted in a relatively efficient way.

https://github.com/apache/iotdb/blob/aaa96b974becf7cc57ef5ac36c3fa9e199423010/server/src/main/java/org/apache/iotdb/db/query/executor/fill/LastPointReader.java#L171-L173

Reply:
Hi, I didn't notice that. If that, maybe use a iterator here to traverse the list is better. In this way we can do these two methods both efficiently.

Reply:
Wow. It is the optimal solution.